ludwig/schema/preprocessing.py

```diff
@@ -0,0 +1,751 @@
+    )
+
+    lowercase: Optional[bool] = schema_utils.Boolean(
+        default=False,
+        description="If true, converts the string to lowercase before tokenizing.",
+    )
+
+    most_common: Optional[int] = schema_utils.PositiveInteger(
+        default=10000,
+        allow_none=True,
+        description="The maximum number of most common tokens to be considered. If the data contains more than this "
+        "amount, the most infrequent tokens will be treated as unknown.",
+    )
+
+
+@dataclass
+@register_preprocessor("sequence")
+class SequencePreprocessingConfig(schema_utils.BaseMarshmallowConfig):
+
+    tokenizer: Optional[str] = schema_utils.String(
+        default="space",
+        allow_none=False,
+        description="Defines how to map from the raw string content of the dataset column to a sequence of elements.",
+    )
+
+    vocab_file: Optional[str] = schema_utils.String(
+        default=None,
+        allow_none=True,
+        description="Filepath string to a UTF-8 encoded file containing the sequence's vocabulary. On each line the "
+        "first string until \t or \n is considered a word.",
+    )
+
+    max_sequence_length: Optional[int] = schema_utils.PositiveInteger(
+        default=256,
+        allow_none=False,
+        description="The maximum length (number of tokens) of the text. Texts that are longer than this value will be "
+        "truncated, while texts that are shorter will be padded.",
+    )
+
+    most_common: Optional[int] = schema_utils.PositiveInteger(
+        default=20000,
+        allow_none=False,
+        description="The maximum number of most common tokens in the vocabulary. If the data contains more than this "
+        "amount, the most infrequent symbols will be treated as unknown.",
+    )
+
+    padding_symbol: Optional[str] = schema_utils.String(
+        default="<PAD>",
+        allow_none=False,
+        description="The string used as a padding symbol. This special token is mapped to the integer ID 0 in the "
+        "vocabulary.",
+    )
+
+    unknown_symbol: Optional[str] = schema_utils.String(
+        default="<UNK>",
+        allow_none=False,
+        description="The string used as an unknown placeholder. This special token is mapped to the integer ID 1 in "
+        "the vocabulary.",
+    )
+
+    padding: Optional[str] = schema_utils.StringOptions(
+        ["left", "right"],
+        default="right",
+        allow_none=False,
+        description="the direction of the padding. right and left are available options.",
+    )
+
+    lowercase: Optional[bool] = schema_utils.Boolean(
+        default=False,
+        description="If true, converts the string to lowercase before tokenizing.",
+    )
+
+    missing_value_strategy: Optional[str] = schema_utils.StringOptions(
+        ["fill_with_const", "fill_with_mode", "fill_with_mean", "backfill", "bfill", "pad", "ffill", "drop_row"],
```

Comment:
Consider factoring out a `UNIVERSAL_MISSING_VALUE_STRATEGIES` which has the missing value strategies that are shared across all features.

Reply:
Will do!

Reply:
Done!